kbuild-0.1.9998.3407/src/kmk/loadapi.c: In function 'gmk_eval':
kbuild-0.1.9998.3407/src/kmk/loadapi.c:62:51: error: passing argument 3 of 'eval_buffer' makes pointer from integer without a cast [-Wint-conversion]
   62 |   eval_buffer (s, flp IF_WITH_VALUE_LENGTH_PARAM (strlen (s) /** @todo suboptimal */));
      |                                                   ^~~~~~~~~~
      |                                                   |
      |                                                   size_t {aka long unsigned int}

eval_buffer expects end-of-string as the last parameter - it was getting just the length for some reason

see also: https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-build/kbuild/files/kbuild-0.1.9998.3499-int-conversion.patch?id=b1a1b5d41844e17d59cc5de373a12fab9a216ca7

diff --git a/src/kmk/loadapi.c b/src/kmk/loadapi.c
index 677bc33..df892f4 100644
--- a/src/kmk/loadapi.c
+++ b/src/kmk/loadapi.c
@@ -59,7 +59,7 @@ gmk_eval (const char *buffer, const gmk_floc *gfloc)
   install_variable_buffer (&pbuf, &plen);
 
   s = xstrdup (buffer);
-  eval_buffer (s, flp IF_WITH_VALUE_LENGTH_PARAM (strlen (s) /** @todo suboptimal */));
+  eval_buffer (s, flp IF_WITH_VALUE_LENGTH_PARAM (s + strlen (s) /** @todo suboptimal */));
   free (s);
 
   restore_variable_buffer (pbuf, plen);
